lib/sign_in.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:badiup/models/user_model.dart';
+import 'package:badiup/constants.dart' as Constants;
+
+final FirebaseAuth _auth = FirebaseAuth.instance;
+final GoogleSignIn googleSignIn = GoogleSignIn();
+User currentSignedInUser;
+
+Future<String> signInWithGoogle() async {
+  final GoogleSignInAccount googleSignInAccount = await googleSignIn.signIn();
+  final GoogleSignInAuthentication googleSignInAuthentication =
+      await googleSignInAccount.authentication;
+
+  final AuthCredential credential = GoogleAuthProvider.getCredential(
+    accessToken: googleSignInAuthentication.accessToken,
+    idToken: googleSignInAuthentication.idToken,
+  );
+
+  final FirebaseUser user = await _auth.signInWithCredential(credential);
+
+  assert(!user.isAnonymous);
+  assert(await user.getIdToken() != null);
+
+  final FirebaseUser currentUser = await _auth.currentUser();
+  assert(user.uid == currentUser.uid);
+
+  // add user to DB, user email as document ID
+  currentSignedInUser = User(
+      name: user.displayName,
+      email: user.email,
+      //isAdmin: false,
+      created: DateTime.now().toUtc(),
+  );
+  Firestore.instance.collection(Constants.DBCollections.USERS).document(user.email).setData(
```

Comment:
Doesn't this function call need to be awaited?

Reply:
True. I will update it.